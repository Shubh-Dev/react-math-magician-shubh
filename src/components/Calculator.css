* {
  text-decoration: none;
  list-style-type: none;
  box-sizing: border-box;
}

body {
  margin: 0;
  padding: 0;
  background: linear-gradient(to right, #eea486, #dfd9e1);
  font: 20px/1.5 Arial, Helvetica, sans-serif;
}

.container {
  margin: 10vh auto 10vh auto;
  display: grid;
  grid-template-columns: repeat(4, 8rem);
  grid-template-rows: minmax(5rem, auto) repeat(5, 6rem);
  justify-content: center;
}

.bigger-btn {
  grid-column: span 2;
}

.display-area {
  grid-column: 1 / -1;
  background-color: rgb(138, 138, 134);
  color: #fff;
  font-size: 30px;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  padding: 26px 5px 0 3px;
}

.container > button {
  cursor: pointer;
  font-size: 2rem;
  border: 1px solid #fff;
  outline: none;
}

.orange-color {
  background-color: orange;
}

.title-cal {
  margin-left: 20px;
}
