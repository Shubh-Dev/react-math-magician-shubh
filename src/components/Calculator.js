import React, { useState } from 'react';
import './Calculator.css';
import calculate from '../logic/Calculate';

const Calculator = () => {
  const [result, setResult] = useState({ total: 0, next: null, operation: null });
  const handleClick = (e) => {
    setResult(calculate(result, e.target.value));
  };
  const { total, next, operation } = result;
  return (
    <>
      <h2 className="title-cal">Let&apos;s do some Math!!</h2>
      <div className="container">
        <div className="display-area">
          {total}

          {operation}

          {next}

        </div>
        <button type="button" value="AC" onClick={handleClick}>
          AC
        </button>
        <button type="button" value="+/-" onClick={handleClick}>
          +/-
        </button>
        <button type="button" value="%" onClick={handleClick}>
          %
        </button>
        <button type="button" className="orange-color" value="รท" onClick={handleClick}>
          รท
        </button>
        <button type="button" value="7" onClick={handleClick}>
          7
        </button>
        <button type="button" value="8" onClick={handleClick}>
          8
        </button>
        <button type="button" value="9" onClick={handleClick}>
          9
        </button>
        <button type="button" className="orange-color" value="x" onClick={handleClick}>
          x
        </button>
        <button type="button" value="4" onClick={handleClick}>
          4
        </button>
        <button type="button" value="5" onClick={handleClick}>
          5
        </button>
        <button type="button" value="6" onClick={handleClick}>
          6
        </button>
        <button type="button" className="orange-color" value="-" onClick={handleClick}>
          -
        </button>
        <button type="button" value="1" onClick={handleClick}>
          1
        </button>
        <button type="button" value="2" onClick={handleClick}>
          2
        </button>
        <button type="button" value="3" onClick={handleClick}>
          3
        </button>
        <button type="button" className="orange-color" value="+" onClick={handleClick}>
          +
        </button>
        <button type="button" className="bigger-btn" value="0" onClick={handleClick}>
          0
        </button>
        <button type="button" value="." onClick={handleClick}>
          .
        </button>
        <button type="button" className="orange-color" value="=" onClick={handleClick}>
          =
        </button>
      </div>
    </>
  );
};
export default Calculator;
