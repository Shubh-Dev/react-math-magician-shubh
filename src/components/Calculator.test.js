import '@testing-library/jest-dom';
import {
  render,
  screen,
} from '@testing-library/react';
import Calculator from './Calculator';

test('check if the Caluclator component rendered by checking if the % button exist in the DOM', () => {
  render(<Calculator />);
  const resultElement = screen.getByText('%', { exact: false });
  expect(resultElement).toBeInTheDocument();
});
