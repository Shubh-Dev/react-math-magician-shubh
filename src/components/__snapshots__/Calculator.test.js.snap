// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculator renders correctly 1`] = `[Function]`;
